src/main/java/nl/fuchsia/services/FeitService.java

```diff
@@ -30,4 +30,25 @@ public Feit addFeit(Feit feit) {
     public List<Feit> getFeiten() {
         return feitRepository.getFeiten();
     }
+
+    public Feit updateFeitById(Feit feit) {
+
+        try {
+            Feit feitOpgehaald = feitRepository.getFeitById(feit.getFeitNr());
+
+            if (feitOpgehaald == null) {
+                throw new NotFoundException("Feitnummer: " + feit.getFeitNr() + " bestaat niet!");
+            }
+            if (!(feitOpgehaald.getFeitcode().equals(feit.getFeitcode()) )){
+                throw new UniekVeldException("Feitcode: " + feitOpgehaald.getFeitcode() + " mag niet gewijzigd worden in " + feit.getFeitcode());
```

Comment:
Wellicht moet de error tekst aangeven dat de feitcode reeds bestaat.

Reply:
Andere velden stonden expliciet in de US, de code niet, dus mag/hoeft niet gewijzigd kunnen worden.

Reply:
Top!